### PR TITLE
 gnrc_sixlowpan_iphc: fix issues found by Coverity scan

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

These two look like typos:

- [Out-of-bounds read](https://scan3.coverity.com/reports.htm#v46910/p10250/fileInstanceId=38359631&defectInstanceId=9793718&mergedDefectId=298279)
- [result_independent_of_operands](https://scan3.coverity.com/reports.htm#v46910/p10250/fileInstanceId=38359631&defectInstanceId=9793843&mergedDefectId=298295)

### Testing procedure

Not sure what tests apply, but this was likely not covered by a test before.

### CODE REVIEW

File: sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c [hunk 1]
```diff
@@ -1035,7 +1035,7 @@ static size_t _iphc_ipv6_encode(gnrc_pktsnip_t *pkt,
 
         /* copy remaining byteos of flow label */
         iphc_hdr[inline_pos++] = (uint8_t)((ipv6_hdr_get_fl(ipv6_hdr) & 0x0000ff00) >> 8);
-        iphc_hdr[inline_pos++] = (uint8_t)((ipv6_hdr_get_fl(ipv6_hdr) & 0x000000ff) >> 8);
+        iphc_hdr[inline_pos++] = (uint8_t)((ipv6_hdr_get_fl(ipv6_hdr) & 0x000000ff) >> 0);
```
Comment (miri64) [inline sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c:1038]: Not really fighting for this. ;-)

```suggestion
        iphc_hdr[inline_pos++] = (uint8_t)(ipv6_hdr_get_fl(ipv6_hdr) & 0x000000ff);
```

File: sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c [hunk 2]
```diff
@@ -1035,7 +1035,7 @@ static size_t _iphc_ipv6_encode(gnrc_pktsnip_t *pkt,
 
         /* copy remaining byteos of flow label */
```
Comment (cgundogan) [inline sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c:1036]:
```suggestion
        /* copy remaining bytes of flow label */
```

while you are at it?

File: sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c [hunk 3]
```diff
@@ -417,7 +417,7 @@ static size_t _iphc_ipv6_decode(const uint8_t *iphc_hdr,
                 ipv6_hdr->dst.u8[1] = iphc_hdr[payload_offset++];
                 ipv6_hdr->dst.u8[2] = iphc_hdr[payload_offset++];
                 ipv6_hdr->dst.u8[3] = ctx->prefix_len;
-                ipv6_addr_init_prefix((ipv6_addr_t *)ipv6_hdr->dst.u8 + 4,
+                ipv6_addr_init_prefix((ipv6_addr_t *)(ipv6_hdr->dst.u8 + 4),
                                       &ctx->prefix, ctx->prefix_len);
                 memcpy(ipv6_hdr->dst.u8 + 12, iphc_hdr + payload_offset + 2, 4);
```
Comment (miri64) [inline sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c:422]: This expresses the following paragraph in [RFC 6282]

>       If M=1 and DAC=1:
> 
>          00:  48 bits.  This format is designed to match Unicast-Prefix-
>             based IPv6 Multicast Addresses as defined in [RFC3306] and
>             [RFC3956].  The multicast address takes the form ffXX:XXLL:
>             PPPP:PPPP:PPPP:PPPP:XXXX:XXXX. where the X are the nibbles
>             that are carried in-line, in the order in which they appear
>             in this format.  P denotes nibbles used to encode the prefix
>             itself.  L denotes nibbles used to encode the prefix length.
>             The prefix information P and L is taken from the specified
>             context.

So :heavy_check_mark: 

[RFC 6282]: https://tools.ietf.org/html/rfc6282#section-3.1.1

File: sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c (hunk 1 above)
Comment (benpicco) [reply sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c:1038]: I thought I'd keep the symmetry, but looks like this very symmetry masked a bug.

**Pull request merged**